pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -204,32 +269,251 @@ func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconci
 		reqLogger.Errorf("Failed to update NodeMaintenance with \"Succeeded\" status. Error: %v", err)
 		return r.reconcileAndError(instance, err)
 	}
+	reqLogger.Infof("Reconcile completed for Node: %s", nodeName)
+
 
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileNodeMaintenance) stopNodeMaintenanceImp(node *corev1.Node ) error {
+	// Uncordon the node
+	err := AddOrRemoveTaint(r.drainer.Client, node, false)
+	if err != nil {
+		return err
+	}
+
+	if err = drain.RunCordonOrUncordon(r.drainer, node, false); err != nil {
+		return err
+	}
+
+	if err := r.invalidateLease(node.Name); err != nil {
+		return err
+    }
+	return nil
+}
+
 func (r *ReconcileNodeMaintenance) stopNodeMaintenance(nodeName string) error {
 	node, err := r.fetchNode(nodeName)
 	if err != nil {
 		return err
 	}
+	return r.stopNodeMaintenanceImp(node)
+}
+
+func (r *ReconcileNodeMaintenance) invalidateLease(nodeName string) error {
+		if r.isLeaseSupported {
+				log.Info("Lease object supported,  invalidating lease")
+
+				nName := apitypes.NamespacedName{Namespace: LeaseNamespace, Name: nodeName}
+				lease := &coordv1beta1.Lease{}
+
+				if err := r.client.Get(context.TODO(), nName, lease); err != nil {
+
+					if  errors.IsNotFound(err) {
+						return nil
+					}
+					return err
+				}
+				lease.Spec.AcquireTime = nil
+				lease.Spec.LeaseDurationSeconds = nil;
```

Comment:
Java called, it wants its semicolon back :wink:

Reply:
done